### PR TITLE
Added .gitallowed and updated Readme accordingly

### PR DESCRIPTION
**Pull request by eperegrine**

### CODE REVIEW

**Comment by eperegrine** [comment]
> Looks good Emily - does .gitallowed-base work with git-secrets for file excludes on Linux as well as Mac?

I haven't tested it on Linux; I'm a bit limited on time I can dedicate this week

`.gitallowed` is the name needed, and base is used in this folder, and then in the readme copy command, it renames it. I named it that way to try and illustrate that it's a base file to expand upon.

File: nhsd-git-secrets/.gitallowed-base [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.*git-secrets/.*:*
+.*terraform.tfstate.*:*'
+.*.js.map:*
+.*.svg:*
+.*.jquery.*.js:*
```
Comment (andyblundell) [inline nhsd-git-secrets/.gitallowed-base:5]: Hi - did you get a lot of false positives in js files? Feels like we might hide some genuine secrets with this exclusion

File: nhsd-git-secrets/scan-repo-linux-mac.sh [hunk 2]
```diff
@@ -7,8 +7,6 @@ export PATH=$PATH:.
 # # These only need to be run once per workstation/slave/agent but are included to try and ensure they are present
 ./git-secrets --register-aws
 ./git-secrets --add-provider -- cat nhsd-git-secrets/nhsd-rules-linux-mac.txt
-./git-secrets --add --allowed '.*git-secrets/.*:*'
```
Comment (andyblundell) [inline nhsd-git-secrets/scan-repo-linux-mac.sh:10]: Great - I thought there was an issue with gitallowed, but if this is working, that's great :)

File: nhsd-git-secrets/.gitallowed-base (hunk 1 above)
Comment (eperegrine) [reply nhsd-git-secrets/.gitallowed-base:5]: The exclusion there is specifically js map files, I thought it was being picked up by numbers that aided mapping but I should've looked closer

In the capion repo it's this file that was picked up: Capion/Capion.Web/Capion.Web/wwwroot/modules/nhsd/dist/cap-nhsd-scripts.bundle.js.map and it seems to be the ip address regex picking up comments similar to this: // Support: Promises/A+ section 2.3.3.3.1 which is in the jquery source code. I ignored jquery as it is a library so wouldn't have secrets we are concerned about

As a map it doesn't contain any code that isn't written anywhere else

**Comment by andyblundell** [comment]
Hi - this all look really good to me - only question is whether we hold fire on merging until the linux compatibility has been checked. Windows compatibility is still outstanding, or at least I think it is - we have another team trying this on Windows at the moment. @JohnEllwoodBJSS are you in a position to give the new version a quick try on linux, if @eperegrine won't get a chance this week?

**Comment by JohnEllwoodBJSS** [comment]
Hi @andyblundell, I was off on Friday but I can do this today

**Review by andyblundell — APPROVED** [review]

**Pull request merged**